crates/plugins/mediator-coordination/src/pickup/model.rs

```diff
@@ -0,0 +1,199 @@
+    pub(crate) newest_received_time: Option<u32>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) oldest_received_time: Option<u32>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) total_bytes: Option<usize>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) live_delivery: Option<bool>,
+}
+
+impl<'a> From<StatusResponse<'a>> for MessageBuilder {
+    fn from(value: StatusResponse<'a>) -> Self {
+        Message::build(
+            value.id.to_owned(),
+            value.type_.to_owned(),
+            json!(value.body),
+        )
+    }
+}
+
+#[derive(Debug, Serialize, Default)]
+pub(crate) struct DeliveryResponse<'a> {
+    pub(crate) id: &'a str,
+
+    pub(crate) thid: &'a str,
+
+    #[serde(rename = "type")]
+    pub(crate) type_: &'a str,
+
+    pub(crate) body: BodyDeliveryResponse<'a>,
+
+    pub(crate) attachments: Vec<Attachment>,
+}
+
+#[derive(Debug, Serialize, Default)]
+pub(crate) struct BodyDeliveryResponse<'a> {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) recipient_did: Option<&'a str>,
+}
+
+impl<'a> From<DeliveryResponse<'a>> for MessageBuilder {
+    fn from(value: DeliveryResponse<'a>) -> Self {
+        Message::build(
+            value.id.to_owned(),
+            value.type_.to_owned(),
+            json!(value.body),
+        )
+        .thid(value.thid.to_owned())
+        .attachments(value.attachments)
+    }
+}
+
+#[derive(Debug, Serialize, Default)]
+pub(crate) struct LiveDeliveryChange<'a> {
+    pub(crate) id: &'a str,
+
+    pub(crate) pthid: &'a str,
+    
+    #[serde(rename = "type")]
+    pub(crate) type_: &'a str,
+
+    pub(crate) body: BodyLiveDeliveryChange<'a>,
+}
+
+#[derive(Debug, Serialize, Default)]
+pub(crate) struct BodyLiveDeliveryChange<'a> {
+    pub(crate) code: &'a str,
+
+    pub(crate) comment: &'a str,
+}
+
+impl<'a> From<LiveDeliveryChange<'a>> for MessageBuilder {
+    fn from(value: LiveDeliveryChange<'a>) -> Self {
+        Message::build(
+            value.id.to_owned(),
+            value.type_.to_owned(),
+            json!(value.body),
+        )
+        .pthid(value.pthid.to_owned())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use didcomm::{MessageBuilder, Attachment};
+    use serde_json::{json, Value};
+
+    #[test]
+    fn test_status_response_serialization() {
+        let id = "123456789";
+        let type_ = "https://didcomm.org/messagepickup/3.0/status-response";
+        let response = StatusResponse {
+            id,
+            type_,
+            body: BodyStatusResponse {
+                recipient_did: Some("did:example:recipient"),
+                message_count: 5,
+                longest_waited_seconds: Some(160),
+                live_delivery: Some(false),
+                ..Default::default()
+            },
+        };
+
+        let message_builder: MessageBuilder = response.into();
+        let message = message_builder.finalize();
```

Comment:
These tests mainly cover the conversion to Message, and serialization is only partial to the bodies.
- First, the name of the test is therefore misleading.
- Second, do you need serialization of the main response object? Then you need to test it, else drop the trait.

Reply:
The message builder is just a builder for the message, it converts to messages. I don’t really get your point here.